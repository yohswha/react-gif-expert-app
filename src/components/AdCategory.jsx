import { useState } from "react"

export const AdCategory = ({onNewCategory}) => {
    const [inputValue, setinputValue] = useState('');

    const onInputChange=({target})=>{
        setinputValue(target.value);
    }
    const onSubmit= (event) => {
        event.preventDefault();
        var inputValor=inputValue.trim();
        if (inputValor.length<=1) return;
        onNewCategory(inputValor);
        setinputValue("");
    }

    return (


        < form onSubmit={(event)=>onSubmit(event)}>
            <input
                type="text"
                placeholder="Buscar Gifts"
                value={inputValue}
                onChange={onInputChange}
            />
        </form >
    )
}
