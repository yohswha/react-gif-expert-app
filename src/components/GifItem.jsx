import propTypes from 'prop-types'

export const GifItem = ({title,url}) => {
    console.log(title)

  return (
    
    <div className="card">
        <img src={url} alt={title}/>
        <p>{title}</p>
    </div>
    
  )
}

GifItem.propTypes={
  title: propTypes.string.isRequired,
  url  : propTypes.string.isRequired
}
