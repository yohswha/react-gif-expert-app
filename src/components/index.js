export * from './AdCategory';
export * from './GifGrid';
export * from './GifItem';