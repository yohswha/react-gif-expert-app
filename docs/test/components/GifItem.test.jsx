import { render } from "@testing-library/react";
import { GifItem } from "../../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => { 
    test('debe hacer match con el snapShot',()=>{
        const title='Saitama';
        const url= 'https://one-punch.com/saitama.jpg';

       const {container}= render(<GifItem title={title} url={url} />);
       expect(container).toMatchSnapshot();

    });
 });