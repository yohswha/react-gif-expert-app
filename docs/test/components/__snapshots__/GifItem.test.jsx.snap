// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapShot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
